### PR TITLE
migrate connector in kafka batch sink

### PR DESCRIPTION
**Pull request by itsankit-google**
[PLUGIN-968](https://cdap.atlassian.net/browse/PLUGIN-968)

### CODE REVIEW

File: kafka-plugins-0.10/docs/Kafka-batchsink.md [hunk 1]
```diff
@@ -12,6 +12,11 @@ compression types to events. This plugin uses kafka 0.10.2 java apis.
 
 Configuration
 -------------
+**Use Connection** Whether to use a connection, if a connection is used,
+the brokers do not need to be provided.
+
+**Connection** Name of the connection to use, should have the macro function ${conn:(connection-name)} to provide.
```
Comment (yaojiefeng) [inline kafka-plugins-0.10/docs/Kafka-batchsink.md]: should be `${conn(connection-name)}`

File: kafka-plugins-0.10/src/main/java/io/cdap/plugin/sink/KafkaBatchSink.java [hunk 2]
```diff
@@ -138,19 +142,23 @@ public void destroy() {
...
-    @Description("Specifies the connection string where Producer can find one or more brokers to " +
-      "determine the leader for each topic")
+    @Name(ConfigUtil.NAME_USE_CONNECTION)
+    @Nullable
+    @Description("Whether to use an existing connection.")
+    private Boolean useConnection;
+
+    @Name(ConfigUtil.NAME_CONNECTION)
     @Macro
-    private String brokers;
```
Comment (yaojiefeng) [inline kafka-plugins-0.10/src/main/java/io/cdap/plugin/sink/KafkaBatchSink.java:153]: This is backward incompatible change that causes the unit test to fail, this is strange as the source and sink have basically same config but with different names, `kafkaBrokers` vs `brokers`. I think what we can do is to create a new `KafkaSinkConnectorConfig` object, which contains both fields, both `@Nullable`, where the `getBrokers` will be based on the config that is not null. In the widget json, make `kafkaBrokers` hidden so user can only see one config.

File: kafka-plugins-0.10/widgets/Kafka-batchsink.json [hunk 3]
```diff
@@ -13,11 +41,11 @@
           "name": "referenceName"
         },
         {
-          "widget-type": "csv",
-          "label": "Kafka Brokers",
-          "name": "brokers",
```
Comment (yaojiefeng) [inline kafka-plugins-0.10/widgets/Kafka-batchsink.json]: This is backward incompatible change, check my previous comment on how we should properly handle it.

File: kafka-plugins-0.10/docs/Kafka-batchsink.md (hunk 1 above)
Comment (itsankit-google) [reply kafka-plugins-0.10/docs/Kafka-batchsink.md]: done

File: kafka-plugins-0.10/src/main/java/io/cdap/plugin/sink/KafkaBatchSink.java (hunk 2 above)
Comment (itsankit-google) [reply kafka-plugins-0.10/src/main/java/io/cdap/plugin/sink/KafkaBatchSink.java:153]: done

File: kafka-plugins-0.10/widgets/Kafka-batchsink.json (hunk 3 above)
Comment (itsankit-google) [reply kafka-plugins-0.10/widgets/Kafka-batchsink.json]: done

**Review by yaojiefeng — APPROVED** [review]

**Pull request merged**